components/infra-proxy-service/server/databags.go

```diff
@@ -65,6 +65,11 @@ func (s *Server) CreateDataBagItem(ctx context.Context, req *request.CreateDataB
 		return nil, err
 	}
 
+	// Custom data params id attribute check
+	if req.Data == nil || len(req.Data.Fields) == 0 || req.Data.Fields["id"].GetStringValue() == "" {
```

Comment:
This looks good to me for now. But we should be able to validate nested structs too in the `Validate` function above.

Reply:
As of now validation handler not handled the nested struct key or other types of key & values, need to make validation handler more flexible to deals these cases.